File: 832-flipping-an-image/832-flipping-an-image.py
class Solution:
    def flipAndInvertImage(self, image: List[List[int]]) -> List[List[int]]:
        p=[]
        for i in image:
            p.append(self.reverse(i))
            
        return p
        
    def reverse(self, list):
        l=0
        r=len(list)-1
        while l<r:
            list[l],list[r]=list[r],list[l]
            if list[l]==0:
                list[l]=1
            else:
                list[l]=0
            if list[r]==0:
                list[r]=1
            else:
                list[r]=0
            l=l+1
            r=r-1
        if len(list)%2==1:
            if list[len(list)//2]==0:
                list[len(list)//2]=1
            else:
                list[len(list)//2]=0
        return list

File: 832-flipping-an-image/832-flipping-an-image.java
class Solution {
    public int[][] flipAndInvertImage(int[][] image) {
        
        for(int i=0;i<image.length;i++) {
            reverse(image[i]); 
            flipNumber(image[i]);   
            
        }
        return image;
    }
    
    public void flipNumber(int[] arr) {
        int l = 0;
        int r = arr.length-1; 
        
        for(int i=0;i<=r;i++) {
            if(arr[i] == 0) {
                arr[i] = 1;
            } else {
                arr[i] = 0;
            }
        }
        
    }
    
    public void reverse(int[] arr) {
        int l = 0;
        int r = arr.length-1; 
        int mid = (l+r)/2;
        while(l<=r) {
            int temp = arr[l];
            arr[l] = arr[r];
            arr[r] = temp;
            l++;
            r--; 
        }   
    }
}